app/controllers/admin/namespaces_controller.rb

```diff
@@ -1,6 +1,10 @@
 class Admin::NamespacesController < Admin::BaseController
+  respond_to :html, :js
+
   def index
     @special_namespaces = Namespace.where(global: true)
-    @namespaces = Namespace.not_portus.where(global: false).page(params[:page])
+    namespaces = Namespace.not_portus.where(global: false)
+    @namespaces = namespaces.search(params[:filter]).page(params[:page])
```

Comment:
I believe it would. First time with `Namespace.not_portus.where` (1st where query) and second time with `namespaces.search` (the [search method](https://github.com/mrkamel/search_cop#how-does-it-work) is also a where query)
If they where on the same assignment, so Rails would know to unify them - but when are split - they would be called one after the other.